.circleci/config.yml

```diff
@@ -156,35 +156,27 @@ jobs:
       
       - run:
           name: Install EvaDB package from GitHub repo with all dependencies
+          environment:
+            PY_VERSION: << parameters.v >>
+            RAY: << parameters.ray >>
           command: |
             "python<< parameters.v >>" -m venv test_evadb
             pip install --upgrade pip
             source test_evadb/bin/activate
-            pip install ".[dev]"
-
-      # Enable Ray (update evadb.yml file and install Ray package)
-      - when:
-          condition:
-            equal: [ ENABLED, << parameters.ray >> ]
-          steps:
-            - run:
-                name: Enable Ray setting in the config.yml file
-                command: |
-                  source test_evadb/bin/activate
-                  python -c "import yaml;f = open('evadb/evadb.yml', 'r+');config_obj = yaml.load(f, Loader=yaml.FullLoader);config_obj['experimental']['ray'] = True;f.seek(0);f.write(yaml.dump(config_obj));f.truncate();"
-                  pip install ".[ray]"
-      
-      # Install qdrant only on versions < 3.11
-      - when:
-          condition:
-            not:
-              equal: [ "3.11", << parameters.v >> ]
-          steps:
-            - run:
-                name: Install Qdrant
-                command: |
-                  source test_evadb/bin/activate
-                  pip install ".[qdrant]"
+            if [ $RAY = "ENABLED" ]; then
+              if [ $PY_VERSION != "3.11" ]; then
+                pip install ".[dev,ray,qdrant]"
+              else
+                pip install ".[dev,ray]"
+              fi
+              python -c "import yaml;f = open('evadb/evadb.yml', 'r+');config_obj = yaml.load(f, Loader=yaml.FullLoader);config_obj['experimental']['ray'] = True;f.seek(0);f.write(yaml.dump(config_obj));f.truncate();"
+            else
+              if [ $PY_VERSION != "3.11" ]; then
+                pip install ".[dev,ludwig,qdrant]"
```

Comment:
@gaurav274 @jiashenC @jarulraj To avoid dependency conflict, I have changed the way that circle ci installs the EvaDB. Let me know if you have a different idea. Thanks.

Reply:
I am fine with this as well. Another way to rewrite this using the current infrastructure is when the ray is disabled and the python version is not 3.11. We already have both of those flags. If I understand correctly, we just need to combine https://github.com/georgia-tech-db/evadb/blob/master/.circleci/config.yml#L168 and https://github.com/georgia-tech-db/evadb/blob/master/.circleci/config.yml#L181?

Reply:
I am not able to find a way to do `when else` in circle ci. I am following the suggestions from https://discuss.circleci.com/t/conditional-steps-if-else-without-code-duplication/45030

Reply:
```
// we already have those two
if ray enabled: install ray
if python != 3.11: install qdrant

// we add for this PR
if ray disabled and python != 3.11: install ludwig 
```

We can use nested logic https://circleci.com/docs/configuration-reference/#logic-statements.